src/ui/public/vislib/vis.js

```diff
@@ -64,8 +64,12 @@ export default function VisFactory(Private) {
         uiState.on('change', this._uiStateChangeHandler = () => this.render(this.data, this.uiState));
       }
 
+      this.resizeChecker.stopSchedule();
```

Comment:
This is a pretty serious collection of steps. What do you think about naming it something and adding it as a method to the `ResizeChecker`. 

Maybe something like:
```js
ResizeChecker.prototype.changeWithoutTriggering = function(block) {
  // prevent checking while the block runs
  this.stopSchedule();
  
  // run the external code that will cause the size to change
  block();

  // save the new size, without emitting a change, and restart the schedule
  this.saveSize();
  this.saveDirty(false);
  this.continueSchedule();
};
```

Which would be used as:
```js
this.resizeChecker.changeWithoutTriggering(() => {
  this.handler = handlerTypes[chartType](this) || handlerTypes.column(this);
  this._runOnHandler('render');
})
```